daemon/src/wallet.rs

```diff
@@ -345,17 +356,113 @@ where
     }
 }
 
+struct LockedUtxos {
+    inner: HashSet<(Instant, OutPoint)>,
+    time_to_lock: Duration,
+}
+
+impl LockedUtxos {
+    /// The minimum time that must pass before a locked UTXO is
+    /// released on Bitcoin mainnet.
+    ///
+    /// We have chosen a value of 20 minutes to allow for 1 or 2
+    /// confirmations if the UTXO was actually used. This way we are
+    /// unlikely to attempt to reuse the same UTXO and compete with
+    /// one of our transactions in the mempool.
+    const TIME_TO_LOCK_MAINNET: Duration = Duration::from_secs(20 * 60);
+
+    /// The minimum time that must pass before a locked UTXO is
+    /// released on any of the Bitcoin test networks.
+    const TIME_TO_LOCK_TEST: Duration = Duration::from_secs(2 * 60);
```

Comment:
@thomaseizinger @da-kami I didn't end up changing these values but I documented the reason why the mainnet values were chosen. Does this make sense to you?